src/prefect/environments/execution/local.py

```diff
@@ -0,0 +1,35 @@
+import os
+
+import prefect
+from prefect.environments.execution.base import Environment
+from prefect.environments.storage import Storage
+
+
+class LocalEnvironment(Environment):
+    """
+    A LocalEnvironment class for executing a flow contained in Storage in the local process.
+    Execution will first attempt to call `get_runner` on the storage object, and if that fails it will
+    fall back to `get_env_runner`.  If `get_env_runner` is used, the environment variables from this
+    process will be passed.
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    def execute(self, storage: "Storage", flow_location: str) -> None:
```

Comment:
I suggest two explicit args:
- `use_env_runner=False` to specify whether to attempt the env runner (there may be Storage classes that implement both)
-  `env=None` to provide environment explicitly for `env_runner`. It won't always be true we want to inherit the local environment.

I think the second suggestion is more important/useful than the first.

Reply:
If possible, I'd prefer to keep the interface static for environments.  If we think those arguments are universally useful / necessary, then I'm cool with adding them to this and all other environments.

> there may be Storage classes that implement both

could you think of an example where this would be the case?

> It won't always be true we want to inherit the local environment.

In my mind, this is the sort of decision that environments make.  So `LocalEnvironment` makes the decision that your local setup is how you want things run.  Moreover, using your current environment guarantees that the flow run has same behavior regardless of whether there is an environment interface or a more direct interface.  Open to discussion on this point.

Reply:
I agree, those `env` related args should be environment specific and I don't think the environments should be generalized out to all support a wide range of patterns. I too prefer keeping them static

Reply:
+1 I'm convinced